src/NuGet.Core/NuGet.Configuration/PackageSource/PackageSourceProvider.cs

```diff
@@ -296,50 +301,31 @@ private static PackageSourceCredential ReadCredentialFromEnvironment(string sour
                 validAuthenticationTypesText: match.Groups["authTypes"].Value);
         }
 
-        private PackageSource GetPackageSource(string key, Dictionary<string, IndexedPackageSource> sourcesLookup)
+        public PackageSource GetPackageSourceByName(string name)
```

Comment:
There are no unit tests for this public method. I acknowledge that this issue wasn't introduced in this PR. I am comfortable with either creating a follow-up issue or adding a simple unit test in this PR, similar to how a unit test was added for the `GetPackageSourceBySource` method.